corehq/apps/users/models.py

```diff
@@ -800,6 +813,8 @@ class CouchUser(Document, DjangoUserMixin, IsMemberOfMixin, UnicodeMixIn, EulaMi
     assigned_location_ids = StringListProperty()
     has_built_app = BooleanProperty(default=False)
 
+    reporting_metadata = SchemaProperty(ReportingMetadata)
```

Comment:
> the reason it's semi-important that this be able to be done in one query is so we can grab just the top 10 users and sort by one of those properties (currently that the reason we have to fetch every user)

I think the way I worked around this in case activity was, if the report is sorted on username then I grabbed  the paginated users from UserES and then added that to a user_id filter for cases. If it was sorted on something else, then I would just add a limit to the case query and look up those user ids in UserES. I think you could do the same here by using last_form_submission.

The resulting query code probably won't be as readable as this, so it'll fit in with the rest of reports :P